src/Entity/ReactionRuleConfig.php

```diff
@@ -160,6 +160,36 @@ public function getExpression() {
   }
 
   /**
+   * Gets the Rules component that is invoked when the events are dispatched.
+   *
+   * The returned component has the definitions of the available event context
+   * set.
+   *
+   * @return \Drupal\rules\Engine\RulesComponent
+   *   The Rules component.
+   */
+  public function getComponent() {
+    $component = RulesComponent::create($this->getExpression());
+    $component->addContextDefinitionsForEvents([$this->getEvent()]);
+    return $component;
+  }
+
+  /**
+   * Sets the Rules component containing the configuration to be stored.
+   *
+   * @param \Drupal\rules\Engine\RulesComponent $component
+   *   The component.
+   *
+   * @return $this
+   */
+  public function setComponent(RulesComponent $component) {
```

Comment:
setComponent()? But no component is set in the implementation? Should this be setComponentConfiguration() or transferComponentConfig() or similar?